### PR TITLE
new panos module to load configuration file

### PR DESCRIPTION
**Pull request by ivanbojer**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
panos_loadcfg

##### ANSIBLE VERSION

```
ansible 2.1.0.0 
config file = 
configured module search path = Default w/o overrides
```

##### SUMMARY
New panos module that allows admin to apply and load new configuration into the firewall. The file has to be imported first using panos_import module.

### CODE REVIEW

File: lib/ansible/modules/network/panos/panos_loadcfg.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      category: "configuration"
+    register: result
+  - name: load configuration
+    panos_loadcfg:
+      ip_address: "192.168.1.1"
+      password: "admin"
+      file: "{{result.filename}}"
+'''
+
+RETURN = '''
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_loadcfg.py]: does not match actual return dict

File: lib/ansible/modules/network/panos/panos_loadcfg.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        username=dict(default='admin'),
+        file=dict(),
+        commit=dict(type='bool', default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    if not HAS_LIB:
+        module.fail_json(msg='pan-python is required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_loadcfg.py]: not needed when `required=True` is specified

File: lib/ansible/modules/network/panos/panos_loadcfg.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        module.fail_json(msg="ip_address should be specified")
+    password = module.params["password"]
+    if not password:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_loadcfg.py]: not needed when `required=True` is specified

File: lib/ansible/modules/network/panos/panos_loadcfg.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    username = module.params['username']
+
+    xapi = pan.xapi.PanXapi(
+        hostname=ip_address,
+        api_username=username,
+        api_password=password
+    )
+
+    file_ = module.params['file']
+    if file_ is None:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_loadcfg.py]: not needed when `required=True` is specified

File: lib/ansible/modules/network/panos/panos_loadcfg.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
+        module.fail_json(msg="ip_address should be specified")
+    password = module.params["password"]
+    if not password:
+        module.fail_json(msg="password is required")
+    username = module.params['username']
+
+    xapi = pan.xapi.PanXapi(
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_loadcfg.py:119]: common function?

**Review by privateip — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/panos/panos_loadcfg.py (hunk 1 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_loadcfg.py]: There is no return value so I mimicked what was done in wakeonlan.py

File: lib/ansible/modules/network/panos/panos_loadcfg.py (hunk 2 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_loadcfg.py]: ok

File: lib/ansible/modules/network/panos/panos_loadcfg.py (hunk 3 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_loadcfg.py]: ok

File: lib/ansible/modules/network/panos/panos_loadcfg.py (hunk 5 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_loadcfg.py:119]: nope, just class instantiation.

File: lib/ansible/modules/network/panos/panos_loadcfg.py (hunk 4 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_loadcfg.py]: ok

**Comment by ivanbojer** [comment]
ready_for_review

**Pull request merged**